lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -401,51 +623,211 @@ class ConvertAten_IndexPutImplOp
       return rewriter.notifyMatchFailure(
           op, "Input element type should be same as the values element type.");
 
-    SmallVector<Value> indicesList;
-    getListConstructElements(adaptor.getIndices(), indicesList);
+    SmallVector<Value> optionalIndicesList;
+    getListConstructElements(op.getIndices(), optionalIndicesList);
     // The size of the list of the index tensors should not be greater than the
     // input rank.
-    if ((int64_t)indicesList.size() > inputType.getRank())
+    if ((int64_t)optionalIndicesList.size() > inputRank)
       return rewriter.notifyMatchFailure(
           op, "Indices list size should not be greater than the input rank.");
 
-    // TODO: Add support for cases with indices list size not equal to 1.
-    if (indicesList.size() != 1)
-      return rewriter.notifyMatchFailure(
-          op, "Unimplemented: Indices list size != 1");
-    Value indexTensor = indicesList[0];
+    Value torchCstNone = rewriter.create<Torch::ConstantNoneOp>(loc);
+    unsigned sizeOptionalIndicesList = optionalIndicesList.size();
+    SmallVector<int64_t> nonNoneIndexTensorDim;
+    unsigned numNonNoneIndices;
 
-    if (indexTensor.getType().isa<Torch::NoneType>())
-      return rewriter.notifyMatchFailure(op, "Index tensor must not be None.");
+    if (sizeOptionalIndicesList == 0) {
+      return rewriter.notifyMatchFailure(op, "Indices list must not be empty.");
+    } else {
```

Comment:
The previous `if` returns early. No need to have `else` branch (decreases indentation)